src/main/java/com/jcabi/ssh/Execution.java

```diff
@@ -0,0 +1,173 @@
+import com.jcraft.jsch.ChannelExec;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Execution of a single command.
+ * @author Georgy Vlasov (wlasowegor@gmail.com)
+ * @version $Id$
+ * @since 1.4
+ */
+interface Execution {
+    /**
+     * Executes some command.
+     * @return Return code of the command.
+     * @throws IOException If fails
+     */
+    int exec() throws IOException;
+
+    /**
+     * Execution of a command in an SSH session.
+     * @author Georgy Vlasov (wlasowegor@gmail.com)
+     * @version $Id$
+     * @since 1.4
+     */
+    final class Default implements Execution {
+        /**
+         * Command.
+         */
+        private final transient String command;
+
+        /**
+         * Stdin.
+         */
+        private final transient InputStream stdin;
+
+        /**
+         * Stdout.
+         */
+        private final transient OutputStream stdout;
+
+        /**
+         * Stderr.
+         */
+        private final transient OutputStream stderr;
+
+        /**
+         * Session.
+         */
+        private final transient Session session;
+
+        /**
+         * Uses an SSH session to execute a single command and disconnect
+         * immediately.
+         * @param cmd Command
+         * @param input Stdin (will be closed)
+         * @param out Stdout (will be closed)
+         * @param err Stderr (will be closed)
+         * @param sess SSH session (will be disconnected)
+         * @checkstyle ParameterNumberCheck (6 lines)
+         */
+        Default(final String cmd, final InputStream input,
+            final OutputStream out, final OutputStream err,
+            final Session sess) {
+            this.command = cmd;
+            this.stdin = input;
+            this.stdout = out;
+            this.stderr = err;
+            this.session = sess;
+        }
+
+        /**
+         * Execute {@link #command} in {@link #session}.
+         * @return Exit code
+         * @throws IOException If fails
+         */
+        @Override
+        public int exec() throws IOException {
+            try {
+                try {
+                    final ChannelExec channel = ChannelExec.class.cast(
+                        this.session.openChannel("exec")
+                    );
+                    channel.setErrStream(this.stderr, false);
+                    channel.setOutputStream(this.stdout, false);
+                    channel.setInputStream(this.stdin, false);
+                    channel.setCommand(this.command);
+                    channel.connect();
+                    Logger.info(this, "$ %s", this.command);
+                    return this.exec(channel);
+                } finally {
+                    this.session.disconnect();
+                }
+            } catch (final JSchException ex) {
```

Comment:
@Suseika I didn't understand why do we need two try/catch constructs here? why can't we just use one?